src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,465 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// The MLPerf Training workload executor.
+    /// </summary>
+    [UnixCompatible]
+    public class MLPerfTrainingExecutor : VirtualClientComponent
+    {
+        private const string AccuracySummary = nameof(MLPerfTrainingExecutor.AccuracySummary);
+        private const string PerformanceSummary = nameof(MLPerfTrainingExecutor.PerformanceSummary);
+
+        private IStateManager stateManager;
+        private ISystemManagement systemManager;
+
+        /// <summary>
+        /// Constructor for <see cref="MLPerfTrainingExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public MLPerfTrainingExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+            this.systemManager = this.Dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManager.StateManager;
+
+            if (string.IsNullOrEmpty(this.DiskFilter))
+            {
+                this.DiskFilter = "SizeGreaterThan:8000gb&OSDisk:false";
+            }
+        }
+
+        /// <summary>
+        /// Disk filter string to filter disks to format.
+        /// </summary>
+        public string DiskFilter
+        {
+            get
+            {
+                // Change the size -----------------------------------------------------------------------------------------------------------------------------
+                string filter = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.DiskFilter), "SizeGreaterThan:1000gb");
+                // Enforce filter to remove OS disk.
+                filter = $"{filter}&OSDisk:false";
+                return filter;
+            }
+
+            set
+            {
+                this.Parameters[nameof(MLPerfTrainingExecutor.DiskFilter)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training model name (e.g. bert, rnnt, ssd-mobilenet).
+        /// </summary>
+        public string Model
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Model));
```

Comment:
Do we not want to keep any default model, because if someone will not provide Model, workload will start failing, will it not be better to have a default model?

Reply:
I have added default model as bert